### PR TITLE
Add ostree-ext-cli container compare image1 image2

### PR DESCRIPTION
**Pull request by RishabhSaini**
Allows to compare the content between two OCI compliant images

### CODE REVIEW

File: lib/src/container/mod.rs [hunk 1]
```diff
@@ -246,6 +247,64 @@ impl std::fmt::Display for OstreeImageReference {
...
+        if !src_layers.contains_key(blobid) {
+            diff.added.push(descriptor.clone());
+        }
+    }
+    diff
+}
+
+impl ManifestDiff {
+    /// Prints the total, removed and added content between two OCI images
+    pub fn print(&self) {
```
Comment (cgwalters) [inline lib/src/container/mod.rs:290]: I go back and forth on whether we should have code that calls `println!` in the library.  In some ways it feels architecturally unclean, but in practice it's...practical :smile:

**Review by cgwalters — APPROVED** [review]
Nice!

**Comment by cgwalters** [comment]
BTW though it would be nice to have some unit/integration tests for this.  Maybe in `priv-integration.sh` to say diff an image with itself and verify it shows zero modified layers to start?

File: lib/src/container/mod.rs (hunk 1 above)
Comment (RishabhSaini) [reply lib/src/container/mod.rs:290]: agreed

**Comment by RishabhSaini** [comment]
> BTW though it would be nice to have some unit/integration tests for this. Maybe in `priv-integration.sh` to say diff an image with itself and verify it shows zero modified layers to start?

https://github.com/ostreedev/ostree-rs-ext/pull/459

**Pull request merged**